### PR TITLE
Show notice when no funds ready for deposit

### LINKED ISSUE #7879: Deposits: It is not clear to merchants that pending balances will not be paid out
### Describe the bug

When a merchant has a $0 or negative available balance it is not clear that the pending balance will not be paid out.

We should add a notice on the deposits overview page with clear wording. Something like 

```
Deposits are paused until your available balance is above $0.
```

related to #7876

### PR DESCRIPTION
**Pull request by brucealdridge**
Fixes #7879

> **Note**
> This branch includes changes from #7875 
> That PR needs to be merged first.

### Changes proposed in this Pull Request

When merchants have a pending balance but no funds are yet available it is not clear that scheduled deposits will not take place.

This PR adds a notice, letting merchants know that funds are not yet available for deposit.

**Notice Text**

> You have no funds available to deposit. [Why?](https://woo.com/document/woopayments/deposits/deposit-schedule/#pending-funds)

Please feedback on the notice copy. I went through several variations but ultimately landed on something simple and attention-grabbing to point merchants to the docs.

@csmcneill and @bradyv y'all are much better at this 😄 

image

### Testing instructions

I was able to get my store in the appropriate state by going into dev-tools and overriding the response of the overview-all request to ensure that available balance was `0` and pending was larger than `0`

image

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7882` or branch name `fix/7879-show-notice-when-no-funds-ready-for-deposit` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/7879-show-notice-when-no-funds-ready-for-deposit) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: d27594409afb51fd3bf81ea163762a887e3ff22b
- Build time: 2023-12-13 05:53:38 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by bradyv** [comment]
@brucealdridge I actually think the copy you've proposed works perfectly. I can't think of any improvements to make!

**Comment by Jinksi** [comment]
@brucealdridge I changed base to the branch from #7875 to clarify changes this PR introduces. Hope that's cool!

File: client/components/deposits-overview/deposit-notices.tsx [hunk 1]
```diff
@@ -150,3 +150,37 @@ export const NegativeBalanceDepositsPausedNotice: React.FC = () => (
...
+/**
+ * Renders a notice informing the user that deposits only occur when there are funds available.
+ */
+export const NoFundsAvailableForDepositNotice: React.FC = () => (
+	<InlineNotice
+		status="warning"
+		icon
+		className="no-funds-available-for-deposit-notice"
+		isDismissible={ false }
+	>
```
Comment (Jinksi) [inline client/components/deposits-overview/deposit-notices.tsx]: Suggestion: remove className that deosn't look like it's being used.
```suggestion
	<InlineNotice status="warning" icon isDismissible={ false }>
```

File: client/components/deposits-overview/index.tsx [hunk 2]
```diff
@@ -47,9 +48,12 @@ const DepositsOverview: React.FC = () => {
 	const isLoading = isLoadingOverview || isLoadingDeposits;
 
 	const availableFunds = overview?.available?.amount ?? 0;
+	const pendingFunds = overview?.pending?.amount ?? 0;
 
 	// If the available balance is negative, deposits may be paused.
 	const isNegativeBalanceDepositsPaused = availableFunds < 0;
+	// When there are funds pending but no available funds, deposits are paused.
+	const isFundsAvailableForDeposit = availableFunds === 0 && pendingFunds > 0;
```
Comment (Jinksi) [inline client/components/deposits-overview/index.tsx]: This variable name is confusing me a bit, implies that there is available balance ready for a deposit.

`hasPendingFundsForFutureDeposit` maybe?

File: client/components/deposits-overview/deposit-notices.tsx [hunk 3]
```diff
@@ -150,3 +150,37 @@ export const NegativeBalanceDepositsPausedNotice: React.FC = () => (
...
+		{ interpolateComponents( {
+			mixedString: sprintf(
+				/* translators: %s: WooPayments */
+				__(
+					'You have no funds available to deposit. {{whyLink}}Why?{{/whyLink}}',
+					'woocommerce-payments'
+				),
+				'WooPayments'
+			),
```
Comment (Jinksi) [inline client/components/deposits-overview/deposit-notices.tsx]: Suggestion: remove redundant sprintf that isn't being used.


```suggestion
		{ interpolateComponents( {
			mixedString: __(
				'You have no funds available to deposit. {{whyLink}}Why?{{/whyLink}}',
				'woocommerce-payments'
			),
```

File: client/components/deposits-overview/index.tsx (hunk 2 above)
Comment (Jinksi) [reply client/components/deposits-overview/index.tsx]: `isDepositAwaitingPendingFunds`?

**Review by Jinksi — APPROVED** [review]
This tests well Bruce! It's a great improvement to help communicate why pending funds are not available for deposit.

I've left some code review comments and suggestions, but this works as-is.

File: client/components/deposits-overview/index.tsx (hunk 2 above)
Comment (brucealdridge) [reply client/components/deposits-overview/index.tsx]: Thanks, I spent way too long on variable names. I think I changed it 4 times, your ideas are much better.

**Comment by brucealdridge** [comment]
> This tests well Bruce! It's a great improvement to help communicate why pending funds are not available for deposit.

Based on feedback and changes, I'm marking as ready to merge, feel free to review again if needed @Jinksi

**Comment by Jinksi** [comment]
Rebased to `fix/7874-show-deposit-schedule-when-zero-available-balance` and removed all commits carried over from `develop`.

**Comment by rogermattic** [comment]
Hey @brucealdridge! Thanks for pointing me to this PR! Like @bradyv  said, the copy you suggested is great. 

Not urgent, but something to keep in mind for future tweaks of this view: The fact that we first say (1)  the balance is $0.0 then (2) that deposit is scheduled for the next day and then (3) again that no funds are available YET there are some funds in Pending. It can be slightly confusing, depending on how merchants digest the hierarchy of information on this screen :) 

Screenshot 2023-12-13 at 12 09 14

**Comment by Jinksi** [comment]
> The fact that we first say (1) the balance is $0.0 then (2) that deposit is scheduled for the next day and then (3) again that no funds are available YET there are some funds in Pending.

@rogermattic thanks for your feedback! We have resolved (2) now by hiding the "your next deposit is scheduled" when there is no available balance.

image

Note: we're also adjusting the 2 banners in the above screenshot to only show one or the other in #7902.

**Comment by rogermattic** [comment]
@Jinksi 

Awesome! Thanks for letting me know!

**Pull request merged**